core/src/main/java/cz/o2/proxima/repository/AttributeValueDescriptor.java

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright 2017-2021 O2 Czech Republic, a.s.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package cz.o2.proxima.repository;
+
+import cz.o2.proxima.scheme.SchemaDescriptors.ArrayTypeDescriptor;
+import cz.o2.proxima.scheme.SchemaDescriptors.PrimitiveTypeDescriptor;
+import cz.o2.proxima.scheme.SchemaDescriptors.SchemaTypeDescriptor;
+import cz.o2.proxima.scheme.SchemaDescriptors.StructureTypeDescriptor;
+import java.io.Serializable;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+
+/**
+ * Attribute value descriptor.
+ *
+ * @param <T> attribute type
+ */
+@EqualsAndHashCode
+public class AttributeValueDescriptor<T> implements Serializable {
+
+  private static final long serialVersionUID = 1L;
+
+  /** Return {@link SchemaTypeDescriptor} for give attribute value */
+  @Getter private final SchemaTypeDescriptor<T> schemaDescriptor;
+
+  private AttributeValueDescriptor(SchemaTypeDescriptor<T> schemaDescriptor) {
+    this.schemaDescriptor = schemaDescriptor;
+  }
+
+  /**
+   * Create {@link AttributeValueDescriptor} with Primitive type.
+   *
+   * @param descriptor primitive type descriptor
+   * @param <T> attribute type
+   * @return Attribute value descriptor
+   */
+  public static <T> AttributeValueDescriptor<T> of(PrimitiveTypeDescriptor<T> descriptor) {
+    return of(descriptor.toTypeDescriptor());
```

Comment:
Looks like these three metods can be turn to one accepting `GenericTypeDescriptor`?

Reply:
👍